gitlab/resource_gitlab_project_freeze_period.go

```diff
@@ -0,0 +1,148 @@
+package gitlab
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	gitlab "github.com/xanzy/go-gitlab"
+)
+
+func resourceGitlabProjectFreezePeriod() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGitlabProjectFreezePeriodCreate,
+		Read:   resourceGitlabProjectFreezePeriodRead,
+		Update: resourceGitlabProjectFreezePeriodUpdate,
+		Delete: resourceGitlabProjectFreezePeriodDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"freeze_start": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"freeze_end": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cron_timezone": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "UTC",
+			},
+		},
+	}
+}
+
+func resourceGitlabProjectFreezePeriodCreate(d *schema.ResourceData, meta interface{}) error {
+	projectID := d.Get("project_id").(string)
+
+	options := gitlab.CreateFreezePeriodOptions{
+		FreezeStart:  gitlab.String(d.Get("freeze_start").(string)),
+		FreezeEnd:    gitlab.String(d.Get("freeze_end").(string)),
+		CronTimezone: gitlab.String(d.Get("cron_timezone").(string)),
+	}
+
+	log.Printf("[DEBUG] Project %s create gitlab project-level freeze period %+v", projectID, options)
+
+	client := meta.(*gitlab.Client)
+	FreezePeriod, _, err := client.FreezePeriods.CreateFreezePeriodOptions(projectID, &options)
+	if err != nil {
+		return err
+	}
+
+	FreezePeriodIDString := fmt.Sprintf("%d", FreezePeriod.ID)
+	d.SetId(buildTwoPartID(&projectID, &FreezePeriodIDString))
+
+	return resourceGitlabProjectFreezePeriodRead(d, meta)
+}
+
+func resourceGitlabProjectFreezePeriodRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*gitlab.Client)
+	projectID, freezePeriodID, err := projectIDAndFreezePeriodIDFromID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] read gitlab FreezePeriod %s/%d", projectID, freezePeriodID)
+
+	freezePeriod, _, err := client.FreezePeriods.GetFreezePeriod(projectID, freezePeriodID)
+	d.Set("id", freezePeriod.ID)
```

Comment:
I think the "id" attribute is reserved. Are tests passing?

Reply:
id should be the unique identifier accross all resources